src/main/java/com/graphhopper/converter/resources/ConverterResourceGisgraphy.java

```diff
@@ -0,0 +1,188 @@
+
+	public static final String REVERSE_QUERY_TYPE = "reverse";
+	public static final String GEOCODING_QUERY_TYPE = "geocoding";
+	public static final String AUTOCOMPLETE_QUERY_TYPE = "autocomplete";
+
+	private final String geocodingUrl;
+	private final String reverseGeocodingUrl;
+	private final String searchURL;
+	private final String apiKey;
+	private final Client jerseyClient;
+
+	public ConverterResourceGisgraphy(String geocodingUrl,
+			String reverseGeocodingUrl, String searchURL, String apiKey,
+			Client jerseyClient) {
+		this.geocodingUrl = geocodingUrl;
+		this.reverseGeocodingUrl = reverseGeocodingUrl;
+		this.searchURL = searchURL;
+		this.jerseyClient = jerseyClient;
+		this.apiKey = apiKey;
+	}
+
+	@GET
+	@Timed
+	public Response handle(@QueryParam("q") @DefaultValue("") String query,
+			@QueryParam("lat") @DefaultValue("") String lat,
+			@QueryParam("lng") @DefaultValue("") String lng,
+			@QueryParam("radius") @DefaultValue("") String radius,
+			@QueryParam("country") @DefaultValue("") String country,
+			@QueryParam("limit") @DefaultValue("5") Integer limit,
+			@QueryParam("querytype") @DefaultValue("geocoding") String queryType) {
+		limit = fixLimit(limit);
+
+		WebTarget target;
+		if (queryType.equalsIgnoreCase(REVERSE_QUERY_TYPE)) {
+			target = buildReverseGeocodingTarget(query, lat, lng, radius,
+					country, limit);
+		} else if (queryType.equalsIgnoreCase(AUTOCOMPLETE_QUERY_TYPE)) {
+			target = buildAutocompleteTarget(query, lat, lng, radius, country,
+					limit);
+		} else {
+			queryType = GEOCODING_QUERY_TYPE;
+			target = buildGeocodingTarget(query, lat, lng, radius, country,
+					limit);
+		}
+
+		Response response = target.request().accept("application/json").get();
+		Status status = new Status(response.getStatus(), response
+				.getStatusInfo().getReasonPhrase());
+		failIfResponseNotSuccessful(target, status);
+
+		if (queryType.equalsIgnoreCase(GEOCODING_QUERY_TYPE)
+				|| queryType.equalsIgnoreCase(REVERSE_QUERY_TYPE)) {
+			GisgraphyGeocodingResult feed = response
+					.readEntity(GisgraphyGeocodingResult.class);
+			if (feed != null) {
+				return Converter.convertFromGisgraphyList(feed.result, status);
+			} else {
+				LOGGER.error("There was an issue with the target "
+						+ target.getUri() + " the provider returned: "
+						+ status.code + " - " + status.message);
+				throw new BadRequestException(
+						"error deserializing geocoding feed");
+			}
+		} else if (queryType.equalsIgnoreCase(AUTOCOMPLETE_QUERY_TYPE)) {
+			GisgraphySearchResult feed = response
+					.readEntity(GisgraphySearchResult.class);
+			if (feed != null) {
+				return Converter.convertFromGisgraphySearchList(feed.getResponse(), status);
+			} else {
+				LOGGER.error("There was an issue with the target "
+						+ target.getUri() + " the provider returned: "
+						+ status.code + " - " + status.message);
+				throw new BadRequestException(
+						"error deserializing geocoding feed");
+			}
+		}
```

Comment:
As you mentioned, I think here should be indeed another else that throws and exception